### PR TITLE
Better implementation of confirmation while quitting Ferdi

### LINKED ISSUE #1857: Disable Ctrl+Q shotrcut that quits Ferdi without any question
**Is your feature request related to a problem? Please describe.**
Sometimes I accidentally press the `Ctrl+Q` keyboard combination, that's immediately closes the Ferdi app without any questions or confirmations, and I can't find the way to disable this shortcut or enable confirmation of this action.

**Describe the solution you'd like**
Will be glad to have a configuration of all keyboard shortcuts, but this is a large task. So will be good have an option to disable this hotkey only, or maybe add confirmation "Are you sure want to exit Ferdi?" dialog.

### PR DESCRIPTION
**Pull request by vraravam**
#### Pre-flight Checklist

1. Please remember that if you are logging a bug for some service that has _stopped working_ or is _working incorrectly_, please log the bug [here](https://github.com/getferdi/recipes/issues)
2. If you are requesting support for a **new service** in Ferdi, please log it [here](https://github.com/getferdi/recipes/pulls)
3. Please remember to read the [self-help documentation](https://github.com/getferdi/ferdi#troubleshooting-recipes-self-help) - in case it helps you unblock yourself for issues related to older versions of recipes that were installed on your machine. (These will get automatically upgraded when you upgrade to the newer versions of Ferdi, but to get new recipes between Ferdi releases, this documentation is quite useful.)
4. Please consider supporting Ferdi!
  👉 https://github.com/sponsors/getferdi
  👉 https://opencollective.com/getferdi/donate
5. Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/getferdi/ferdi/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/getferdi/ferdi/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [issue tracker](https://github.com/getferdi/ferdi/issues) for a feature request that matches the one I want to file, without success.

#### Description of Change
Re-implemented the confirmation dialog popup such that it works consistently for all use-cases (more than #1879)

#### Motivation and Context
Fixes #1857 

#### Checklist

- [x] My pull request is properly named
- [x] The changes respect the code style of the project (`npm run prepare-code`)
- [x] `npm test` passes
- [x] I tested/previewed my changes locally

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -551,19 +552,33 @@ ipcMain.on('stop-find-in-page', (e, action) => {
...
-app.on('before-quit', () => {
-  willQuitApp = true;
+app.on('before-quit', (event) => {
+  const yesButtonIndex = 0;
+  let selection = yesButtonIndex;
+  if (retrieveSettingValue('confirmOnQuit')) {
+    selection = dialog.showMessageBoxSync(app.mainWindow, {
+      type: 'question',
+      message: 'Quit',
+      detail: 'Do you really want to quit Ferdi?',
```
Comment (mhatvan) [inline src/index.js:570]: Needs intl.

File: src/index.js [hunk 2]
```diff
@@ -551,19 +552,33 @@ ipcMain.on('stop-find-in-page', (e, action) => {
...
     debug('Window: all windows closed, quit app');
     app.quit();
   } else {
     debug("Window: don't quit app");
   }
 });
 
-app.on('before-quit', () => {
-  willQuitApp = true;
+app.on('before-quit', (event) => {
```
Comment (mhatvan) [inline src/index.js:563]: This solution is much better!

**Review by mhatvan — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (vraravam) [reply src/index.js:570]: Unfortunately, `intl` is not yet available at the point this file is loaded. All other hard-coded strings need to be handled for i18n.

Comment (vraravam) [reply src/index.js:570]: Can we take this as a TODO items and move forward? I will add a TODO at the top of the file

**Pull request merged**